keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -335,34 +335,47 @@ The versions of CNI plugin binaries that must be used for proper dual-stack func
 ### Services
 
 #### Type NodePort, LoadBalancer, ClusterIP
-Services will only be assigned an address from a single address family (either IPv4 or IPv6). The address family for the Service’s cluster IP is determined by setting the field, `.spec.ipFamily`, on that Service. You can only set this field when creating a new Service. Setting the `.spec.ipFamily` field is optional and should only be used if you plan to enable IPv4 and IPv6 Services and Ingresses on your cluster. The configuration of this field not a requirement for egress traffic. The default address family for your cluster is the address family of the first service cluster IP range configured via the `--service-cluster-ip-range` flag to the kube-controller-manager. The `.spec.ipFamily` field can be set to either IPv4 or IPv6.
+Services will only be assigned an address from a single address family (either IPv4 or IPv6). This assignment is user controlled by setting the value of `.spec.ipFamily` on that Service. Setting the `.spec.ipFamily` field is optional and should only be used when there are multiple entries configured in `--services-cluster-ip-range` flag. The `.spec.ipFamily` field allows mutability under the same conditions `.spec.ClusterIP` field allows it. Another field will be introduced in Service is `status.preferredIPFamily` field. The value of this field is an array of `ipFamily` where `status.preferredIPFamily[0]` is used by endpoints controller to select the appropriate endpoints. The value of this field is set according to the following rules:
```

Comment:
Naming nits:

lists are usually named in plural so `preferredIPFamilies` which is semantically weird.  Why not just `ipFamilies` which parallels `ipFamily` in capitalization and is shorter and can still be described as "sorted by preference".

Reply:
I agree.